sdk/core/core-client/src/serviceClient.ts

```diff
@@ -426,13 +432,13 @@ export interface ClientPipelineOptions extends InternalPipelineOptions {
  */
 export function createClientPipeline(options: ClientPipelineOptions = {}): Pipeline {
   const pipeline = createPipelineFromOptions(options ?? {});
-
   const credential = options.credentialOptions?.credential;
-  if (credential) {
+  const scopes = options.credentialOptions?.credentialScopes;
+  if (credential && scopes) {
```

Comment:
Since `createClientPipeline` is a public method that convenience clients can call directly, did you want to worry about checking scopes is defined when credential is here too? Or should we figure since this is more of an advanced usage, we don't have to validate again?